packages/devtools_app/lib/src/debugger/hover.dart

```diff
@@ -0,0 +1,153 @@
+// Copyright 2021 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart';
+
+import '../theme.dart';
+
+/// Returns the word in the [line] for the provided hover [dx] offset given
+/// the [line]'s [textStyle].
+String wordForHover(double dx, TextSpan line, TextStyle textStyle) {
+  final index = _hoverIndex(dx, line, textStyle);
+  var word = '';
+  if (index == -1) return word;
+  word = line.children[index].toPlainText();
+  word = _mergedLinkedWords(word, index, line);
+  return word;
+}
+
+/// Merges words linked by a `.`.
+///
+/// For example, hovering over `bar` in `foo.bar.baz` would return
+/// `foo.bar.baz`.
+String _mergedLinkedWords(String word, int index, TextSpan line) {
+  var left = index - 1;
+  while (left > 1) {
+    final prev = line.children[left].toPlainText();
+    final prevprev = line.children[left - 1].toPlainText();
+    if (prev == '.') {
+      word = '$prevprev$prev$word';
```

Comment:
nit: one way to write this logic a little bit more simply is to create a
List<String> that is the components of the line.
Then you can just track the the start and end indexes and use String.join to get the output string when you are done.